From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Mon, 10 Jan 2022 16:53:14 -0500
Subject: [PATCH] cmake: Work around wrong Python extension suffix when
 cross-compiling.

The Python extension suffix retrieved from the EXT_SUFFIX sysconfig
variable is currently incorrent when cross-compiling for pypy (although
it is correct for cpython, at least when using crossenv). Until we see
pypy distributed with proper sysconfig handling when cross-compiling, it
is easier to work around the issue by just specifying a ".so" suffix.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 host/python/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/host/python/CMakeLists.txt b/host/python/CMakeLists.txt
index 3bf5c81dc..f195ffe20 100644
--- a/host/python/CMakeLists.txt
+++ b/host/python/CMakeLists.txt
@@ -38,6 +38,8 @@ add_library(pyuhd SHARED
 # python expects extension modules with a particular suffix
 if(WIN32)
     set_target_properties(pyuhd PROPERTIES PREFIX "lib" SUFFIX ".pyd")
+elseif(CMAKE_CROSSCOMPILING)
+    set_target_properties(pyuhd PROPERTIES PREFIX "lib" SUFFIX ".so")
 else()
     execute_process(
         COMMAND "${PYTHON_EXECUTABLE}" -c
-- 
2.34.1

